### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by nicksmider**
Please review carefully.

After #25 is approved, will merge into this branch before merging back to master

User service is accessed through userActions. Login and logout are working. 

Login takes username and password. -> sends request to backend and stores token in AsyncStorage

Logout takes nothing. -> removes token from storage

Addresses #10

### CODE REVIEW

File: screens/Login/Login.js [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
+    dispatch(userActions.login(this.state.email, this.state.password));
+  }
+
+  async logout(e) {
+    const { dispatch } = this.props.navigation;
+    dispatch(userActions.logout());
+  }
+
+  async getUser(e) {
+    console.log(await userService.getUser());
```
Comment (nicksmider) [inline screens/Login/Login.js]: This is temporary, will remove direct service use here

File: screens/Login/Login.js [hunk 2]
```diff
@@ -18,17 +43,37 @@ class Login extends React.Component {
...
+          buttonStyle={style.loginButton}
+          onPress={this.logout}
+          title="LOG OUT"
+        />
+
+        <Button
+          buttonStyle={style.loginButton}
+          onPress={this.getUser}
+          title="CHECK TOKEN"
+        />
```
Comment (nicksmider) [inline screens/Login/Login.js]: check token and logout buttons temporary to show token storage with login. recommend keeping in until we can test with real backend

File: services/user.service.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ username, password }),
+  };
+
+  let user = {
+    token: 'inaibfiubau8y0e8h',
+  };
+
+  // For when successfully implemented
+  //user = await fetch(AUTHENTICATION, requestOptions);
```
Comment (nicksmider) [inline services/user.service.js]: will be used when we have access to backend

**Review by ajhenry — COMMENTED** [review]
Looks good

**Pull request merged**